crates/sui-network/src/state_sync/mod.rs

```diff
@@ -496,6 +497,22 @@ where
                 .collect::<Vec<_>>();
         }
 
+        // If this is the last checkpoint of a epoch, we need to make sure
+        // new committee is in store before we verify newer checkpoints in next epoch.
+        // This could happen before this validator's reconfiguration finishes, because
+        // state sync does not reconfig.
+        if let Some(EndOfEpochData {
```

Comment:
What's the reason that this is not already covered by `WriteStore::insert_checkpoint`?

Reply:
It doesn't seem ideal that we have to do this in two separate paths..

Reply:
it's actually covered there, but consensus does not call `WriteStore::insert_checkpoint`. I chose to not call it here either because some insert also happened in consensus (checkpoint_store::insert_certified_checkpoint).

Reply:
> it's actually covered there, but consensus does not call `WriteStore::insert_checkpoint`. I chose to not call it here either because some insert already happened in consensus (checkpoint_store::insert_certified_checkpoint).

Reply:
Not blocking but would be good to add a TODO there to eventually find a way to dedup the code

Reply:
done